python/cudf/cudf/core/column/string.py

```diff
@@ -618,10 +620,10 @@ def extract(
         2    <NA>
         dtype: object
         """  # noqa W605
-        if flags != 0:
-            raise NotImplementedError("`flags` parameter is not yet supported")
+        if not _is_supported_regex_flags(flags):
+            raise ValueError("invalid `flags` parameter value")
```

Comment:
Isn't this still a `NotImplementedError`?
```python
raise NotImplementedError(f"`flags` {flags} is/are not yet supported")
```

Reply:
This is what I was told to use as per other functions where this flag was added (in this file).

Reply:
https://github.com/rapidsai/cudf/pull/9482#discussion_r739287787

Reply:
@bdice what was your reasoning for preferring a ValueError to a NotImplementedError? These do appear to be more like NotImplementedErrors but I'm just glancing through.

Reply:
I think my previous suggestion was based on the error text. `"invalid `flags` parameter value"` sounded like a `ValueError` to me at the time. On a second reading, I see why this should be a `NotImplementedError` in context, but I would recommend changing the error message to say "unsupported value" instead of "invalid value." My apologies for the mistake.

Reply:
For reference, here's the docs for NotImplementedError: https://docs.python.org/3/library/exceptions.html#NotImplementedError

Emphasis added:

> This exception is derived from [RuntimeError](https://docs.python.org/3/library/exceptions.html#RuntimeError). In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method, **or while the class is being developed to indicate that the real implementation still needs to be added.**